File: gmall0523realtime/src/main/scala/com/gmall/realtime/util/MyPropertiesUtil.scala
package com.gmall.realtime.util

import java.io.InputStreamReader
import java.nio.charset.StandardCharsets
import java.util.Properties

/**
  * Created with IntelliJ IDEA.
  *
  * @Author: bizh
  * @Date: 2021/4/28 9:51
  * @Description: 读取配置文件
  */
object MyPropertiesUtil {

  /*def main(args: Array[String]): Unit = {

    val properties: Properties = MyProperties.load("config.properties")
    val res: String = properties.getProperty("kafka.broker.list")
    println(res)
  }*/


  /**
  * @Description: 读取配置文件
  * @Param: [propertisName]
  * @return: java.util.Properties
  * @Author: bizh
  * @Date: 2021/4/28
  */
  def load(propertisName:String): Properties = {

    val properties: Properties = new Properties()

    properties.load(new InputStreamReader(Thread.currentThread().getContextClassLoader.getResourceAsStream(propertisName), StandardCharsets.UTF_8))

    properties
  }

}

File: locate-data-filter/src/main/java/com/bzh/bigdata/util/MyPropertiesUtil.java
package com.bzh.bigdata.util;

import java.io.IOException;
import java.io.InputStreamReader;
import java.nio.charset.StandardCharsets;
import java.util.Properties;

/**
 * Created with IntelliJ IDEA.
 *
 * @Author: bizh
 * @Date: 2021/5/19 15:31
 * @Description: 读取配置文件
 */
public class MyPropertiesUtil {

    /*public static void main(String[] args) {
        Properties properties = MyPropertiesUtil.load("config.properties");
        String sqlStr = properties.getProperty("data.highway.sqlstring");
        System.out.println(sqlStr);
    }*/

    public static Properties load(String propertisName) {
        Properties properties = new Properties();

        try {
           properties.load(new InputStreamReader(Thread.currentThread().getContextClassLoader().getResourceAsStream(propertisName), StandardCharsets.UTF_8));
        } catch (IOException e) {
            e.printStackTrace();
            throw new RuntimeException("读取配置文件失败！");
        }
        return properties;

    }
}
